staging/cert-manager-setup/templates/v1upgrade.yaml

```diff
@@ -0,0 +1,113 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: cert-manager-upgrade-backup
+  annotations:
+    "helm.sh/hook": pre-install
```

Comment:
do we need `pre-upgrade` with all the `pre-install`s?

Reply:
yeah i think we do, pre-upgrade also here right?

Reply:
Backup is only for pre-install, since it needs a delete upgrade strategy. Helm doesn't support upgrading since it does deployment stuff outside of helm. If you were upgrading manually (outside of kubeaddons), you'd first do a helm delete, then let the script backup the resources and delete the CRDs.